environment.py

```diff
@@ -424,10 +425,12 @@ def env() -> Mapping[str, Optional[str]]:
         #
         'AZUL_TDR_SOURCES': None,
 
-        # The storage location of the BigQuery datasets for the TDR sources,
-        # e.g. `us-central1`. All sources must reside in the same region to
-        # facilitate slot purchases for efficient reindexing.
-        'AZUL_TDR_SOURCE_LOCATION': None,
+        # Location of BigQuery slots purchased when indexing TDR snapshots, e.g.
+        # `us-central1`. Catalogs that index snapshots with storage locations
+        # that do not match this variable must set `needs_slots` to False. As
+        # many snapshots as possible should reside in this region to take
+        # advantage of slot purchases for efficient reindexing.
```

Comment:
nit might want to add `All sources within a catalog must be within the same location`

Reply:
I don't think that's true. If a catalog's sources are spread across multiple locations, then its not possible for them all to match `AZUL_TDR_SLOT_LOCATION`. Therefore, the catalog must have `needs_slots` set to `False`, and no verification of the sources' locations would be performed.

Reply:
Ah that's a good point, we can just turn off the slots within the catalog config.